botocore/response.py

```diff
@@ -80,6 +80,51 @@ def read(self, amt=None):
             self._verify_content_length()
         return chunk
 
+    def __iter__(self):
+        """Return an iterator to yield lines from the raw stream.
+        """
+        # Reading 1024 bytes at a time seems a sane choice [citation needed].
+        # If the user needs to specify something different they can always call
+        # streaming_body._iter_lines(<other-chunk-size>)
+        default_chunk_size = 1024
+        return self._iter_lines(default_chunk_size)
+
+    def _iter_lines(self, chunk_size):
```

Comment:
What are people's thoughts on exposing this method directly as `iter_lines()` instead of having the default behavior of `__iter__` be line based?

This would be similar to what requests exposes via it's [iter_lines](https://github.com/boto/botocore/blob/develop/botocore/vendored/requests/models.py#L705) method.  Looking at what they provide for `__iter__`, it appears to be chunks of 128 bytes.  That's closer to how I'd expect a streaming response object to behave, though I'd bump up the chunks to maybe 1k or 4k.

Reply:
+1 to the default iterator being larger chunks (4k but that's just me) and but also being able to get lines explicitly via `iter_lines()`.